### ISSUE TITLE
rpfwd stop does not close listening port.

### ISSUE THREAD

**Issue by timukas** [MythicAgents/poseidon#52]
Hi,

Using latest Mythic with latest poseidon agent on Linux.

**rpfwd {"action":"start","port":9000,"remote_ip":"192.168.XX.XX","remote_port":80}** - works as it should. Poseidon agent starts tcp listener on port 9000 and i'm able to use port forwarding.

But strange thing happens when i stop port forwarding **rpfwd {"action":"stop","port":9000}**.
I'm not able to use port forwarding (this is fine), but poseidon agent still listening on port 9000.

Shouldn't listening port (9000) disappear after i stop rpfwd on port 9000?

**Comment by its-a-feature** [comment]
Interesting. The agent should definitely stop listening on that port locally on the target host as well. I'll investigate what might be causing it. Thank you for reporting!

**Comment by timukas** [comment]
More information about 'rpfwd'

On 'victim1' machine i start default ELF x64 poseidon payload.
From Mythic UI i start rpfwd on port 8001 with reverse connection to 192.168.114.220:22 (SSH).
'victim1' netstat shows port 8001:

`tcp        0      0 0.0.0.0:8001            0.0.0.0:*               LISTEN      29921/poseidon_01.e`

Now i connect "**ssh -p 8001 victim1**". SSH connection is ok - i'm able to login (and run ssh commands) and 'victim1' shows listening and established connections:

```
tcp        0      0 0.0.0.0:8001            0.0.0.0:*               LISTEN      29921/poseidon_01.e
tcp        0      0 192.168.115.118:8001    192.168.114.44:48082    ESTABLISHED 29921/poseidon_01.e

```
Mythic shows also correct info from 'Proxies' menu:

![image](https://github.com/MythicAgents/poseidon/assets/1915618/27a5d967-b5d5-4bc5-8123-78f17f0f8bae)


So far all works as it should.

While previous rpfwd on port 8001 is active, now i start second rpfwd on same 'victim1' machine, but different local port (4567) and new destination 192.168.99.100:443 (SSL).

`rpfwd {"action":"start","port":4567,"remote_ip":"192.168.99.100","remote_port":443}`

Now on 'victim1' machine listening port 8001 disappeared. Only port 4567 is listening:

`tcp        0      0 0.0.0.0:4567            0.0.0.0:*               LISTEN      29921/poseidon_01.e`

And connection to new port cannot be established.

So only first rpfwd connection works properly. Once you create seconds/third rpfwd, none of them are working. Even the first 'rpfwd' is not working anymore.

**Comment by its-a-feature** [comment]
Ah, I see the confusion. Mythic currently only supports one rpfwd and one socks _per_ callback. Currently, when sending data back from the agent to Mythic the format doesn't include the port number. I think that's why things might be getting messed up with Poseidon. I'll look at the poseidon code and see about extending the data format to include another identifier to allow multiple socks/rpfwd on a single callback

**Comment by its-a-feature** [comment]
Ok, If you pull the latest Mythic (v3.2.7) and the latest Poseidon (v2.0.17), then you should be able to use multiple rpfwd ports within a single callback. Can you use those two versions and let me know if that fixes it for you?

**Comment by timukas** [comment]
Tested with latest Mythic & Poseidon. 

**rpfwd** now works as it should. I was able to start several port forwarders and use them simultaneously.

Stopping/restarting forwarders from split callback also works.

Thanks for fixing it!

p.s. Noticed, that stopping/starting forwarders from search operation 'Proxies' page does not work.

And link to specific task is broken. Currently it has '/new/tasks/19', but should be '/new/task/19'.
![image](https://github.com/MythicAgents/poseidon/assets/1915618/a793a7d9-35e3-4b6e-8e37-4a93f6523d3f)

**Comment by its-a-feature** [comment]
Glad it's working! I just made a push that should fix that link. I also updated the description for that dialog that pops up when you start/stop from that page. That page isn't actually issuing any tasks to your agent, it's simply starting/stopping ports/tracking that Mythic controls. So for something like SOCKS, it would close/open that port on the Mythic server. For something like rpfwd, all it does it internally mark that session as over so no more data will properly flow through it. It doesn't actually issue any task to your agent to start/stop.
